cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -307,6 +307,16 @@ class SimpleArray
         }
     }
 
+    explicit SimpleArray(small_vector<size_t> const & shape, small_vector<size_t> const & stride, std::shared_ptr<buffer_type> const & buffer)
+        : SimpleArray(buffer)
+    {
+        if (buffer)
+        {
+            m_shape = shape;
+            m_stride = stride;
+        }
+    }
```

Comment:
I create this function according to the function `explicit SimpleArray(small_vector<size_t> const & shape, std::shared_ptr<buffer_type> const & buffer)`, but I'm not sure whether it would create the potential bug without checking `nbytes == buffer->nbytes()`.